src/interfaces/vault.ts

```diff
@@ -304,6 +306,125 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
     ).then(arr => arr.flat());
   }
 
+  /**
+   * Fetch the token amount that has been allowed to be used for deposits
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @returns TokenAllowance
+   */
+  async getDepositAllowance(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address
+  ): Promise<TokenAllowance> {
+    const spenderAddress = await this.getDepositSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.allowance(accountAddress, tokenAddress, spenderAddress);
+  }
+
+  /**
+   * Fetch the token amount that has been allowed to be used for withdraw
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @returns TokenAllowance
+   */
+  async getWithdrawAllowance(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address
+  ): Promise<TokenAllowance> {
+    const spenderAddress = await this.getWithdrawSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.allowance(accountAddress, vaultAddress, spenderAddress);
+  }
+
+  /**
+   * Approve the token amount to allow to be used for deposits
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @param amount
+   * @param overrides
+   * @returns TransactionResponse
+   */
+  async approveDeposit(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address,
+    amount?: Integer,
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
+    const spenderAddress = await this.getDepositSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.approve(
+      accountAddress,
+      tokenAddress,
+      spenderAddress,
+      amount ?? MaxUint256.toString(),
```

Comment:
This seems dangerous as the nullish coalescing operator returns the right-hand side operand when its left-hand side operand is `null` or `undefined` otherwise returns its left-hand side operand. What are we trying to do here?

```typescript
console.log(false ?? "10")
// false
```

Reply:
Yeah, that is the intended use. When amount is not provided and therefore is `undefined`, we use `MaxUint256.toString()`. `false` is not possible in this case since amount is not a `boolean`

Reply:
What about when the amount is empty string?
```typescript
console.log("" ?? "256") // will print empty string
```

Reply:
We are expecting a valid `Integer` as an amount. I do understand that cases where we can receive any string, and unfortunately it will break, since we are not validating any inputs, in any place of the sdk.

Reply:
If you look at that type, you'll see it's just an alias for a `string`
```typescript
export type Integer = string;
```
I can give an empty string to amount and it will be happy

Reply:
Totally happy with all the empty strings :)

<img width="418" alt="Screenshot_31_3_2022__11_53_am" src="https://user-images.githubusercontent.com/78794805/160954677-a8d1cd7a-b68e-49ae-8bd4-b5274344d5a6.png">

Reply:
yes exactly, with the nullish coalescing operator we are just using the right hand side explicitly when the amount is undefined, since we assume all strings sent in either of all params are valid. To have proper validation, it would probably be best to implement Value Objects and use serializers, but that is a more complex design we just dont have now.